tests/components/tplink/test_fan.py

```diff
@@ -33,8 +35,12 @@ async def test_fan_unique_id(hass: HomeAssistant) -> None:
     fan = _mocked_device(modules=[Module.Fan], alias="my_fan")
     with _patch_discovery(device=fan), _patch_connect(device=fan):
         await async_setup_component(hass, tplink.DOMAIN, {tplink.DOMAIN: {}})
-        await hass.async_block_till_done()
+        await hass.async_block_till_done(wait_background_tasks=True)
```

Comment:
This could use a comment why it's necessary to pass `wait_background_tasks`.

Reply:
I typically always do this in tests as I've experienced intermittent issues without it and I think it's good practice. https://github.com/home-assistant/core/pull/112726. I can add a comment referencing the 112726.

Reply:
Comment added to `setup_platform_for_device` in init.